### PR TITLE
Add workflow for bitcoind integration tests

### PR DESCRIPTION
**Pull request by kantai**

### CODE REVIEW

File: .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM rust:buster
+
+WORKDIR /src
+
+COPY . .
+
+RUN cargo test --no-run --workspace
+
+RUN cd / && wget https://bitcoin.org/bin/bitcoin-core-0.20.0/bitcoin-0.20.0-x86_64-linux-gnu.tar.gz
```
Comment (jcnelson) [inline .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests:9]: Is this binary statically-linked?

Comment (kantai) [reply .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests:9]: Yep

**Comment by kantai** [comment]
I'm not actually convinced that it's a good idea to use GH actions for testing:

https://github.community/t/github-actions-cancel-redundant-builds-not-solved/16025

This means that this job (I'm guessing it's like a ~15 minute job) would run on every push to every PR, and subsequent pushes wouldn't cancel the prior pushes. That's pretty annoying.

**Comment by diwakergupta** [comment]
https://github.com/marketplace/actions/cancel-workflow-action  seems promising?

**Comment by kantai** [comment]
> https://github.com/marketplace/actions/cancel-workflow-action seems promising?

Yep -- that appears to work well. I was initially scared off by the fact that it needed a github API token, but the default actions token has enough permissions to cancel workflow runs, so it works pretty seamlessly without a lot of permissions.

**Pull request merged**